diff --git a/velox/experimental/cudf/exec/ToCudf.cpp b/velox/experimental/cudf/exec/ToCudf.cpp
index ad25fa474..a80dba38f 100644
--- a/velox/experimental/cudf/exec/ToCudf.cpp
+++ b/velox/experimental/cudf/exec/ToCudf.cpp
@@ -90,13 +90,11 @@ bool CompileState::compile() {
     return driverFactory_.consumerNode;
   };

-  const bool isParquetConnectorRegistered =
-      facebook::velox::connector::getAllConnectors().count("test-parquet") > 0;
-  auto isTableScanSupported =
-      [isParquetConnectorRegistered](const exec::Operator* op) {
-        return isAnyOf<exec::TableScan>(op) && isParquetConnectorRegistered &&
-            cudfTableScanEnabled();
-      };
+  auto isTableScanSupported = [](const exec::Operator* op) {
+    return isAnyOf<exec::TableScan>(op) &&
+        CudfOptions::getInstance().isParquetConnectorRegistered() &&
+        CudfOptions::getInstance().cudfTableScan;
+  };

   auto isFilterProjectSupported = [](const exec::Operator* op) {
     if (auto filterProjectOp = dynamic_cast<const exec::FilterProject*>(op)) {
@@ -358,8 +356,4 @@ bool cudfDebugEnabled() {
   return FLAGS_velox_cudf_debug;
 }

-bool cudfTableScanEnabled() {
-  return CudfOptions::getInstance().cudfTableScan;
-}
-
 } // namespace facebook::velox::cudf_velox
diff --git a/velox/experimental/cudf/exec/ToCudf.h b/velox/experimental/cudf/exec/ToCudf.h
index 04ec3eeaa..eb458889a 100644
--- a/velox/experimental/cudf/exec/ToCudf.h
+++ b/velox/experimental/cudf/exec/ToCudf.h
@@ -62,6 +62,14 @@ class CudfOptions {
     return prefix_;
   }

+  void setParquetConnectorRegistered(bool newValue) {
+    parquetConnectorRegistered_ = newValue;
+  }
+
+  const bool isParquetConnectorRegistered() const {
+    return parquetConnectorRegistered_;
+  }
+
   const bool cudfEnabled;
   const std::string cudfMemoryResource;
   const bool cudfTableScan;
@@ -75,10 +83,12 @@ class CudfOptions {
         cudfMemoryResource(FLAGS_velox_cudf_memory_resource),
         cudfTableScan(FLAGS_velox_cudf_table_scan),
         memoryPercent(50),
-        prefix_("") {}
+        prefix_(""),
+        parquetConnectorRegistered_(false) {}
   CudfOptions(const CudfOptions&) = delete;
   CudfOptions& operator=(const CudfOptions&) = delete;
   std::string prefix_;
+  bool parquetConnectorRegistered_;
 };

 /// Registers adapter to add cuDF operators to Drivers.
@@ -94,8 +104,9 @@ bool cudfIsRegistered();
 bool cudfDebugEnabled();

 /**
- * @brief Returns true if the velox_cudf_table_scan flag is set to true.
+ * @brief Returns true if an application indicated that the parquet connector is
+ * registered
  */
-bool cudfTableScanEnabled();
+bool isParquetConnectorRegistered();

 } // namespace facebook::velox::cudf_velox